src/pybind/mgr/rook/rook_ceph_client/Readme.md

```diff
@@ -0,0 +1,7 @@
+Files in here are automatically generated by 
```

Comment:
what is this file for?

also i see we have couple "automatically generated" in this PR. is there anyway we can avoid adding them? or generate them at build time?

Reply:
> also i see we have couple "automatically generated" in this PR. is there anyway we can avoid adding them? or generate them at build time?

Thing is, Rook's API is changing quite often and no one is fixing mgr/rook if the API changed. 
I'm open for suggestions of how to not fall behind changes in Rook's API. My idea ist:

1. [Add](https://github.com/rook/rook/pull/3534) JSON-schema validations to Rook's API
2. generate Python model classes out of those validations
3. Use Rook's API in a type-safe way so that we can verify we're *not* falling behind changes in Rook.

We're not going to catch behavior changes that are not reflected in the API definition, but still. 

---

Regarding not committing them. One idea would be to generate those at build time, but then we have to commit the schema definitions here?

Reply:
@tchaikov another idea would be to make https://github.com/sebastian-philipp/rook-ceph-client-python a real project, build a package out of it and then depend on it?